Gemfile.lock

```diff
@@ -3,17 +3,23 @@ GEM
   specs:
     addressable (2.8.6)
       public_suffix (>= 2.0.2, < 6.0)
+    asciidoctor (2.0.22)
+    asciidoctor-diagram (2.3.0)
+      asciidoctor (>= 1.5.7, < 3.x)
+      asciidoctor-diagram-ditaamini (~> 1.0)
+      asciidoctor-diagram-plantuml (~> 1.2021)
+      rexml
+    asciidoctor-diagram-ditaamini (1.0.3)
+    asciidoctor-diagram-plantuml (1.2024.3)
     colorator (1.1.0)
     concurrent-ruby (1.2.3)
     em-websocket (0.5.3)
       eventmachine (>= 0.12.9)
       http_parser.rb (~> 0)
     eventmachine (1.2.7)
-    eventmachine (1.2.7-java)
-    eventmachine (1.2.7-x64-mingw32)
     ffi (1.16.3)
     forwardable-extended (2.6.0)
-    google-protobuf (4.26.1)
+    google-protobuf (4.26.1-arm64-darwin)
```

Comment:
Hmmm doesn't seem right that we are referencing arch specific things here, especially not darwin. Do we really need `Gemfile.lock` under source control?

Reply:
Do we need it? No, strictly speaking we don't. Should we keep it there anyway? Well, according to the bottom of [this page](https://bundler.io/guides/faq.html) in the official Bundler docs:

> Q: Should I commit my `Gemfile.lock` when writing a gem?
> A: Yes, you should commit it. The presence of a `Gemfile.lock` in a gem’s repository ensures that a fresh checkout of the repository uses the exact same set of dependencies every time. We believe this makes repositories more friendly towards new and existing contributors. Ideally, anyone should be able to clone the repo, run `bundle install`, and have passing tests. If you don’t check in your `Gemfile.lock`, new contributors can get different versions of your dependencies, and run into failing tests that they don’t know how to fix.
> 
> Q: But I have read that gems should not check in the Gemfile.lock!
> A: The main advantage of not checking in your Gemfile.lock is that new checkouts (including CI) will immediately have failing tests if one of your dependencies changes in a breaking way. Instead of forcing every fresh checkout (and possible new contributor) to encounter broken builds, the Bundler team recommends either using a tool like [Dependabot](https://dependabot.com/) to automatically create a PR and run the test suite any time your dependencies release new versions.

TL;DR: **yes**, we should probably keep `Gemfile.lock` in Git, as it's what Bundler says to do and I'm inclined to agree with their reasoning. _However_ I really don't like having arch-specific deps listed in a file meant for everyone, so I'm going to see what I can do to fix that now.

Reply:
I wouldn't sink to much time into it if thats the official advice. 

It also make me question my standard habit of `bundle update && bundle exec jekyll serve`

Maybe I should just be doing `bundle exec jekyll serve` or there is some other `bundle` command I should use rather than update...

Reply:
`bundle install` or just `bundle` is what I would use. Unless you're specifically playing with deps, I wouldn't use `bundle update`.

I'm usually running `bundle && bundle exec jekyll serve --livereload` (unless I'm changing things in the `Gemfile` or `_config.yml`, in which case I leave out `--livereload` and rebuild manually after each change because otherwise they don't always get picked up)